### ISSUE TITLE
Error 403 on Forerunner 965

### ISSUE THREAD

**Issue by alivespirit** [SylvainGa/garmin-tesla#5]
Hi,
For some reason app shows just 403 Forbidden error when opened. Glance works fine and shows correct info. But opening Glance shows `Retrieving data`, then main app screen for a millisecond, and then `Error 403: Forbidden`. No idea why - login went fine, no MFA is used, all the default parameters. Tesla app works fine on the phone, as well as Tessie.
Do you have any ideas what could cause this?
Thanks!

**Comment by SylvainGa** [comment]
Hi, sorry for the delay. I haven't logged into github since a few weeks ago.

What does the Glance line show? Does it show your vehicle name? Has it worked before and it's not working anymore or is this a first time installation? To troubleshoot, are you willing to sideload a debug copy of the app that is going to log some info while running? Beside the vehicle name, there is no PII in the log file.

**Comment by alivespirit** [comment]
Hi @SylvainGa, no worries :)
Glance shows correct vehicle name and remaining battery percentage, range etc. It's the first time I've installed an app. I can install a debug version (although I'll need some instruction, till now I've only installed custom watchfaces manually by dropping them to `Garmin/Apps/` in the watch storage).

**Comment by SylvainGa** [comment]
It's basically the same for widgets and apps, although once you unplug the watch, the file disappears from the app folder (unlike watchfaces) but it remains somewhere else on the watch that we have no access to.

Ok. This is more odd if you get access to the range and battery in Glance, which means the app does have access to the car. It will be interesting to see what the log will show. I'll build a version for your 965 and email it to you later on tonight. Is it you who sent me an email about this yesterday?

**Comment by SylvainGa** [comment]
Hmm, the person that sent me an email with the 403 error has a Approach S70, so obviously not you. I've put the debug version here for your Forerunner 965 here

https://github.com/SylvainGa/Tesla-Link/blob/V7.16.0-Debug/Executables/TESLALINK.PRG

Simply click on the 'Download' button to download it to your computer. It's better if you uninstall the ConnectIQ copy of the app first, then connect your watch to your computer and copy the file to the Garmin/Apps folder. Since you can't edit settings in a sideloaded copy, the default will apply. It's normal that you'll have to relogin. 

The log file is limited to 6 KB and it will fill up pretty fast. Once filled, it will be renamed to .BAK and a new one will be created, so only a max of 12 KB of logs is possible, so once you get the 403, stop the watch and grab the log files. Before it can start to log anything though, you'll have to create an empty file called TeslaLink.TXT in the Garmin/Apps/LOG folder. This is the file (and its .BAK if there) that you'll need to send me. You can send it to me through the 'Contact developer' link in ConnectIQ. Not sure if you can send file right away that way but I'll reply and my email address will be shown, just reply and attach the file. Like I mention, no PII is in the log file and if shown, only the first 10 or 20 bytes (out of 128) of the token is shown, so it's safe.

**Comment by alivespirit** [comment]
I've installed debug version and prepared logs, waiting for you to reply on email to send those.
Few things to add: I'm in Ukraine and using Ukrainian SIM-card, and here default wake through API call is not working (here we need to send SMS to vehicle to wake it if needed, otherwise it wakes up by itself for a few minutes every hour). According to logs it seems app doesn't always understand if vehicle is awake and tries to wake it regardless and probably that call gives 403. E.g. this part in the logs:
```
14:48:44 : onReceiveAwake: 403
14:48:44 : stateMachine: vehicle_state online _need_wake true _wake_done false
14:48:44 : stateMachine: Waking vehicle
```
At this point I was standing near the car, it was unlocked and awake (according to Tessie it was definitely awake - shown as awake and I was able to control stuff from the app).
So possibly that switch `_need_wake` doesn't consider `vehicle_state online`?

**Comment by SylvainGa** [comment]
Could be, I'll take a look. See my previous email. I would like to see what the log from the Glance code.Le mer. 6 déc. 2023, à 08 h 13, Roman M. ***@***.***> a écrit :
I've installed debug version and prepared logs, waiting for you to reply on email to send those.
Few things to add: I'm in Ukraine and using Ukrainian SIM-card, and here default wake through API call is not working (here we need to send SMS to vehicle to wake it if needed, otherwise it wakes up by itself for a few minutes every hour). According to logs it seems app doesn't always understand if vehicle is awake and tries to wake it regardless and probably that call gives 403. E.g. this part in the logs:
14:48:44 : onReceiveAwake: 403
14:48:44 : stateMachine: vehicle_state online _need_wake true _wake_done false
14:48:44 : stateMachine: Waking vehicle

At this point I was standing near the car, it was unlocked and awake (according to Tessie it was definitely awake - shown as awake and I was able to control stuff from the app).
So possibly that switch _need_wake doesn't consider vehicle_state online?
